libdrgn/python/program.c

```diff
@@ -1040,3 +1051,51 @@ Program *program_from_pid(PyObject *self, PyObject *args, PyObject *kwds)
 	}
 	return prog;
 }
+
+ThreadIterator *Program_threads(Program *self)
+{
+	ThreadIterator *iterator =
+		(ThreadIterator *)ThreadIterator_type.tp_alloc(
+			&ThreadIterator_type, 0);
+	if (!iterator)
+		return NULL;
+	Py_INCREF(self);
+	iterator->prog = self;
+	struct drgn_error *err = drgn_thread_iterator_create(&self->prog, &iterator->iterator);
+	if (err)
+		return set_drgn_error(err);
+	return iterator;
+}
+
+Thread *Program_crashed_thread(Program *self)
+{
+	struct drgn_thread *drgn_thread;
+	struct drgn_error *err;
+
+	err = drgn_program_crashed_thread(&self->prog, &drgn_thread);
+	if (err)
+		return set_drgn_error(err);
+
+	return Thread_wrap(drgn_thread, self);
+}
+
+Thread *Program_thread(Program *self, PyObject *args, PyObject *kwds)
+{
+	static char *keywords[] = { "tid", NULL };
+	struct drgn_error *err;
+	struct index_arg tid = {};
+	struct drgn_thread *drgn_thread;
+
+	if (!PyArg_ParseTupleAndKeywords(args, kwds, "O&", keywords, index_converter,
+					 &tid))
+		return NULL;
+
+	err = drgn_program_find_thread(&self->prog, tid.uvalue, &drgn_thread);
+	if (err)
+		return set_drgn_error(err);
+	if (drgn_thread == NULL)
+		return (Thread *)PyErr_Format(PyExc_LookupError,
```

Comment:
Rather than casting here, I'd prefer if `Thread_wrap` and its callers return `PyObject *` instead. Then you only need to cast in `Thread_wrap` itself.